sklearn/exceptions.py

```diff
@@ -135,7 +135,14 @@ class FitFailedWarning(RuntimeWarning):
     ...         gs.fit(X, y)  # This will raise a ValueError since C is < 0
     ...     except ValueError:
     ...         pass
-    ...     print(repr(w[-1].message))
+    ...     msg = repr(w[-1].message)
+    ...     # if msg.find('Traceback (most recent call last):\\n') > -1:
+    ...     ct = msg.count('\\n')
+    ...     if ct > 1 or msg.count('\\\\n') > 1:  # has stacktrace
```

Comment:
If you use r' style strings this will be easier to work with

Reply:
Yes that sounds good.  I will make that change.  It took a while to get this working due to some idiosyncrasies in pylatest now escaping new lines where before they were not.  Unfortunately maintaining compatibility with all Python versions makes the code look rather ridiculous and rather than detecting versions, this dynamic handling seems ideal.  Admittedly, it complicates things and this is probably why the traceback was never included before.  Simple but annoying code to carefully do the message filter is always needed which deals with the details of newlines.

Reply:
I can't get r style strings to work any differently.  The unescaping timing is quite tricky here as its going through doc check and Python with different syntax and verification rules both peculiar around white space, that its easiest to leave it as is.  I am not sure the quality can be improved as of now and this may be ideal.

Reply:
Update: Looks like I got them to work, its a marginal 1 character saving removing two backslashes and adding the 'r'